operators/pkg/tenant-controller/tenant_controller.go

```diff
@@ -144,29 +147,60 @@ func (r *TenantReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctr
 		retrigErr = err
 	}
 
-	// update resource quota in the status of the tenant after checking validity of workspaces.
-	tn.Status.Quota = forge.TenantResourceList(workspaces, tn.Spec.Quota)
+	// Determine if the personal namespace should be deleted
 
 	nsName := fmt.Sprintf("tenant-%s", strings.ReplaceAll(tn.Name, ".", "-"))
-	nsOk, err := r.createOrUpdateClusterResources(ctx, &tn, nsName)
-	if nsOk {
-		klog.Infof("Namespace %s for tenant %s updated", nsName, tn.Name)
-		tn.Status.PersonalNamespace.Created = true
-		tn.Status.PersonalNamespace.Name = nsName
-		if err != nil {
-			klog.Errorf("Unable to update cluster resource of tenant %s -> %s", tn.Name, err)
-			retrigErr = err
-			tnOpinternalErrors.WithLabelValues("tenant", "cluster-resources").Inc()
-		}
-		klog.Infof("Cluster resourcess for tenant %s updated", tn.Name)
-	} else {
-		klog.Errorf("Unable to update namespace of tenant %s -> %s", tn.Name, err)
-		tn.Status.PersonalNamespace.Created = false
-		tn.Status.PersonalNamespace.Name = ""
-		retrigErr = err
-		tnOpinternalErrors.WithLabelValues("tenant", "cluster-resources").Inc()
+
+	keepNsOpen := true; // keepNsOpen defines if we should close the personal namespace based on the last login date
+	                    // must be initialized for later use
+
+	// If the personalNamespace is not created and the lastLogin is blank, then this is the first time
+	// the tenant has logged in; update the lastLogin date to the year 2000 (i.e. do not cause any updates)
+	if tn.Status.PersonalNamespace.Created==false && tn.Spec.LastLogin.IsZero() {
+
+	  tn.Spec.LastLogin = metav1.Time{Time: time.Date(2000, 0, 0, 0, 0, 0, 0, time.Now().Location())}
+      klog.Infof("First login of %s: setting lastLogin value of %s in tenant spec.", tn.Name, tn.Spec.LastLogin)
+
```

Comment:
This is not required.

Reply:
I would disagree; I had noticed that removing this part makes the reconciler have problems. This is because the LastLogin, when blank and not set to a particular value, causes some kind of error that causes the reconciler to run over and over again in a short amount of time

Reply:
Ah okay, I think I see your point now. Removing the entire if-else brackets, I agree with that!

Reply:
Resolved in [588c1cd](https://github.com/netgroup-polito/CrownLabs/pull/821/commits/588c1cdc9b1a4bc06229a004b4a7bed433f0a5ce)